docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,493 @@
+- **GCPManagedCluster**: this presents the properties needed to provision and manage the general GCP operating infrastructure for the cluster (i.e project, networking, iam). It would contain similar properties to **GCPCluster** and its reconciliation would be very similar.
+
+```go
+type GCPManagedControlPlaneSpec struct {
+// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+// +optional
+ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
+
+// AddonsConfig defines the addons to enable with the GKE cluster.
+// +optional
+AddonsConfig *AddonsConfig `json:"addonsConfig,omitempty"`
+
+// Logging contains the logging configuration for the GKE cluster.
+// +optional
+Logging *ControlPlaneLoggingSpec `json:"logging,omitempty"`
+
+// EnableKubernetesAlpha will indicate the kubernetes alpha features are enabled
+// +optional
+EnableKubernetesAlpha bool
+
+...
+}
+```
+
+```go
+type GCPManagedClusterSpec struct {
+// Project is the name of the project to deploy the cluster to.
+Project string `json:"project"`
+
+// The GCP Region the cluster lives in.
+Region string `json:"region"`
+
+// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+// +optional
+ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+// NetworkSpec encapsulates all things related to the GCP network.
+// +optional
+Network NetworkSpec `json:"network"`
+
+// FailureDomains is an optional field which is used to assign selected availability zones to a cluster
+// FailureDomains if empty, defaults to all the zones in the selected region and if specified would override
+// the default zones.
+// +optional
+FailureDomains []string `json:"failureDomains,omitempty"`
+
+// AdditionalLabels is an optional set of tags to add to GCP resources managed by the GCP provider, in addition to the
+// ones added by default.
+// +optional
+AdditionalLabels Labels `json:"additionalLabels,omitempty"`
+
+...
+}
+```
+
+**Pros**
+
+- Clearer separation between the lifecycle management of the general cloud infrastructure required for the cluster and the actual managed control plane  (i.e. GKE in this example)
+- Follows the original intentions of an “infrastructure” and “control-plane” provider
+- Enables removal/addition of properties for a managed kubernetes that may be different compared to an unmanaged kubernetes
+- Works with ClusterClass
+
+**Cons**
+
+- Duplication of API definitions between GCPCluster and GCPManagedCluster and reconciliation for the infrastructure cluster
+- Users need to be aware to use GCPManagedCluster instead of GCPCluster when creating a GKE based cluster.
```

Comment:
I don't necessarily see this as a con, but as an explicit expression for intent, which is usually a good thing.

Reply:
agreed. Will update the proposal.